### ISSUE TITLE
Support wms_endpoint imagery

### ISSUE THREAD

**Issue by grischard** [openstreetmap/iD#4828]
Once #4814 gets merged, we should also support ELI sources of the wms_endpoint type [like this one](https://raw.githubusercontent.com/osmlab/editor-layer-index/gh-pages/sources/europe/lu/Luxembourg-GeoportailOpenDataWMS.geojson). The URLs are for getCapabilities requests on a WMS, which return a list of layers and sublayers.

In the iD interface, they could be shown as a folder, maybe with the icon.

**Comment by bhousel** [comment]
Cool!  I labelled as bluesky, but it #4633 might already contain a lot of the code needed.  It's on my list of things to get to..

**Comment by grischard** [comment]
If we're doing this, we should also support WMTS sources. https://josm.openstreetmap.de/wiki/ImageryCompare shows that JOSM has about 100 that ELI doesn't have.

**Comment by bhousel** [comment]
> has about 100 that ELI doesn't have.

Be careful comparing the datasets on count alone.  Lots of the imagery in their in their index is stuff like people's hobby projects, like maps of scotland from the 1800s or whatever.

We also host a proxy on openstreetmap.us that may be able to proxy wmts?  I don't know but  @jonahadkins and @iandees would be able to say.

**Comment by grischard** [comment]
Indeed, it's the wet finger approach, but those wmts entries tend to be catalogs of open data.

Yeah, mapproxy would be one way of doing it, but then we have no idea when another layer is added to the wmts. In general, it would be good to support those useful catalogs.

**Comment by matkoniecz** [comment]
> Be careful comparing the datasets on count alone. Lots of the imagery in their in their index is stuff like people's hobby projects, like maps of scotland from the 1800s or whatever.

To be more specific. Main imagery for Poland is no longer available as WMS, WMTS works ( Geoportal2 WMS is showing as white tiles 99,99% percent of time for some reason https://github.com/osmlab/editor-layer-index/pull/877 ).

As result primary and best imagery for Poland is inaccessible in iD/Vespucci but remains available in JOSM because it supports WMTS.

2023 edit: this is now fixed

**Comment by CharlesMaZ** [comment]
We have very good aerial photos from the geoportal available in Poland. Unfortunately, WMS works slowly and even ceases to be supported. Please support WMTS in iD, because it will enable easier and faster editing and will eliminate problems with using WMS.

**Comment by rmikke** [comment]
:+1:

**Comment by swiss-knight** [comment]
Follow up on the new Open Government Data provided by our Federal Office of Topography and this article https://sosm.ch/welcome-swisstopo/ I would be glad to use OGC WMTS services in iD !    

Because WMS is... well, not super friendly imho. And the TSM scheme is not available for the data I'd like to use as a custom background.    

I was also totally successful using some of these WMTS service in JOSM in no time. ;-)

**Comment by cicku** [comment]
https://github.com/openstreetmap/iD/blob/bc0c1895f7b240a35e1beb6b9220eb15d8c36a70/modules/renderer/background_source.js#L123

Please consider improving this logic here (adding support of reprojection/CRS:84). Currently my state has several sources without EPSG:3857 support at anytime, but they support CRS:84. The real "EPSG" being used is 2272, a state-specific NAD83 system.

**Comment by Artoria2e5** [comment]
@cicku it looks like there's also some 4326 projection code that isn't being made available through the url-only "custom" interface. They may need a new selection box to enter a user-defined `source.projection`.

there *is* some swapping code on 4326 from #7557, so if that selection gets added, CRS:84 support should be basically automatic.

**Comment by grischard** [comment]
@tyrasd could you please put this on your radar?
